packages/gatsby/src/bootstrap/create-graphql-runner.ts

```diff
@@ -28,7 +28,9 @@ export const createGraphQLRunner = (
   }
 ): Runner => {
   // TODO: Move tracking of changed state inside GraphQLRunner itself. https://github.com/gatsbyjs/gatsby/issues/20941
-  let runner = new GraphQLRunner(store, { graphqlTracing })
+  let runner: GraphQLRunner | undefined = new GraphQLRunner(store, {
+    graphqlTracing,
+  })
```

Comment:
Why create it here? And not just let the function itself do it?

```suggestion
  let runner: GraphQLRunner | undefined
```

Reply:
is this typescript `: GraphQLRunner | undefined` definitionhere? 
an thats `let runner = new GraphQLRunnerwithout(...)` without?

i got crazy about this line `undefined = `

should we extract the Type `GraphQLRunner | undefined` in its own type so this code is little more clear what it do?